### PR TITLE
Remove es6-promise as a dependency, include note about compat polyfills

### LINKED ISSUE #79: Remove Promise polyfill
Instead of bringing in a Promise polyfill, this library should encourage users to bring their own if they need to support old browsers, but shouldn't make this assumption by itself.

In my case I do want that polyfill, but I'm already bringing my own, so this one would just unnecessarily increase my bundle size.

### PR DESCRIPTION
**Pull request by rcchen**
Fixes #79 

`make prod` before:
```
                  Asset       Size  Chunks             Chunk Names
  clipboard-polyfill.js    13.1 kB       0  [emitted]  main
                DT.d.ts  250 bytes          [emitted]
clipboard-polyfill.d.ts  543 bytes          [emitted]
```

`make prod` after:
```
                  Asset       Size  Chunks             Chunk Names
  clipboard-polyfill.js    4.73 kB       0  [emitted]  main
                DT.d.ts  250 bytes          [emitted]
clipboard-polyfill.d.ts  543 bytes          [emitted]
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ Get the source using one of the following:
 - Download [`build/clipboard-polyfill.js`](https://raw.githubusercontent.com/lgarron/clipboard-polyfill/master/build/clipboard-polyfill.js) and include it using a `<script>` tag.
 - `npm install clipboard-polyfill` and import as `clipboard`.
 
+Please note that this library uses ES6 features, notably [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), so if you need to support environments that do not natively support `Promise` (ie. IE <= 11) you will need to include that dependency separately. Recommendations include [es6-promise](https://github.com/stefanpenner/es6-promise) and [core-js](https://github.com/zloirock/core-js).
```
Comment (lgarron) [inline README.md]: I specifically avoid using ES6 features and anything that doesn't work in IE. Promises are the only exception, because the API format relies on it. Could you rephrase to reflect that?

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 *.js
 *.js.map
+*.log
```
Comment (lgarron) [inline .gitignore]: I strongly prefer separating functionally unrelated changes in history. Would you mind making a separate PR or at least putting it in a separate commit?

Comment (rcchen) [reply .gitignore]: Fair, new PR at https://github.com/lgarron/clipboard-polyfill/pull/81

**Comment by abraham** [comment]
Would love to see this get merged. I'd prefer not to ship any extra polyfills.
